django/db/models/functions/json.py

```diff
@@ -9,25 +9,36 @@ class JSONArray(Func):
     function = "JSON_ARRAY"
     output_field = JSONField()
 
+    def __init__(self, *expressions, absent_on_null=False):
+        self.absent_on_null = absent_on_null
+        super().__init__(*expressions)
+
     def as_sql(self, compiler, connection, **extra_context):
         if not connection.features.supports_json_field:
             raise NotSupportedError(
                 "JSONFields are not supported on this database backend."
             )
+        if self.absent_on_null and not connection.features.supports_json_absent_on_null:
+            raise NotSupportedError(
+                "ABSENT ON NULL is not supported by this database backend."
+            )
         return super().as_sql(compiler, connection, **extra_context)
 
     def as_native(self, compiler, connection, *, returning, **extra_context):
-        # PostgreSQL 16+ and Oracle remove SQL NULL values from the array by
-        # default. Adds the NULL ON NULL clause to keep NULL values in the
-        # array, mapping them to JSON null values, which matches the behavior
-        # of SQLite.
-        null_on_null = "NULL ON NULL" if len(self.get_source_expressions()) > 0 else ""
+        # Providing the ON NULL clause when no source expressions are provided is a
+        # syntax error on some backends.
+        if len(self.get_source_expressions()) == 0:
+            on_null_clause = ""
+        elif self.absent_on_null:
+            on_null_clause = "ABSENT ON NULL"
+        else:
+            on_null_clause = "NULL ON NULL"
 
         return self.as_sql(
             compiler,
             connection,
             template=(
-                f"%(function)s(%(expressions)s {null_on_null} RETURNING {returning})"
+                f"%(function)s(%(expressions)s {on_null_clause} RETURNING {returning})"
```

Comment:
So this is a VERY minor quibble, but if you don't have any source expressions, the generate SQL will end up having a triple space in it.

If the spaces were moved to lines 33 and 35, and then removed from the template here, it would resolve that, at the expense of making it slightly awkward right here.

Is that even something worth discussing?

Reply:
if I were hand-writing the SQL, I would write it like `JSON_OBJECT(RETURNING JSONB)`, but the ORM will emit `JSON_OBJECT(   RETURNING JSONB)`